File: api/src/main/java/com/ruby/rubia_server/core/controller/TemplateEnhancementController.java
package com.ruby.rubia_server.core.controller;

import com.ruby.rubia_server.core.dto.EnhanceTemplateDTO;
import com.ruby.rubia_server.core.dto.EnhancedTemplateResponseDTO;
import com.ruby.rubia_server.core.dto.SaveTemplateWithAIMetadataDTO;
import com.ruby.rubia_server.core.dto.MessageTemplateRevisionDTO;
import com.ruby.rubia_server.core.service.TemplateEnhancementService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/template-enhancement")
@RequiredArgsConstructor
@Slf4j
public class TemplateEnhancementController {

    private final TemplateEnhancementService templateEnhancementService;

    @PostMapping("/enhance")
    public ResponseEntity<EnhancedTemplateResponseDTO> enhanceTemplate(
            @Valid @RequestBody EnhanceTemplateDTO request) {
        
        log.info("POST /api/template-enhancement/enhance - Enhancing template for company: {}", request.getCompanyId());
        
        try {
            EnhancedTemplateResponseDTO response = templateEnhancementService.enhanceTemplate(request);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            log.error("Error enhancing template: {}", e.getMessage(), e);
            throw e;
        }
    }

    @PostMapping("/save-with-ai-metadata")
    public ResponseEntity<MessageTemplateRevisionDTO> saveTemplateWithAIMetadata(
            @Valid @RequestBody SaveTemplateWithAIMetadataDTO request) {
        
        log.info("POST /api/template-enhancement/save-with-ai-metadata - Saving template with AI metadata for template: {}", request.getTemplateId());
        
        try {
            MessageTemplateRevisionDTO response = templateEnhancementService.saveTemplateWithAIMetadata(request);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            log.error("Error saving template with AI metadata: {}", e.getMessage(), e);
            throw e;
        }
    }
}

File: api-dotnet/Rubia.Server/Controllers/TemplateEnhancementController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Rubia.Server.DTOs;
using Rubia.Server.Services.Interfaces;

namespace Rubia.Server.Controllers;

[ApiController]
[Route("api/template-enhancement")]
[Authorize]
public class TemplateEnhancementController : ControllerBase
{
    private readonly ITemplateEnhancementService _templateEnhancementService;
    private readonly ILogger<TemplateEnhancementController> _logger;

    public TemplateEnhancementController(
        ITemplateEnhancementService templateEnhancementService,
        ILogger<TemplateEnhancementController> logger)
    {
        _templateEnhancementService = templateEnhancementService;
        _logger = logger;
    }

    [HttpPost("enhance")]
    public async Task<ActionResult<EnhancedTemplateResponseDto>> EnhanceTemplate([FromBody] EnhanceTemplateDto request)
    {
        try
        {
            if (!ModelState.IsValid)
            {
                return BadRequest(ModelState);
            }

            _logger.LogInformation("POST /api/template-enhancement/enhance - Enhancing template for company: {CompanyId}", request.CompanyId);

            var response = await _templateEnhancementService.EnhanceTemplateAsync(request);
            return Ok(response);
        }
        catch (ArgumentException ex)
        {
            _logger.LogWarning(ex, "Invalid request for template enhancement");
            return BadRequest(new { message = ex.Message });
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error enhancing template: {Error}", ex.Message);
            return StatusCode(500, "Internal server error");
        }
    }

    [HttpPost("save-with-ai-metadata")]
    public async Task<ActionResult<MessageTemplateRevisionDto>> SaveTemplateWithAIMetadata([FromBody] SaveTemplateWithAiMetadataDto request)
    {
        try
        {
            if (!ModelState.IsValid)
            {
                return BadRequest(ModelState);
            }

            _logger.LogInformation("POST /api/template-enhancement/save-with-ai-metadata - Saving template with AI metadata for template: {TemplateId}", request.TemplateId);

            var response = await _templateEnhancementService.SaveTemplateWithAIMetadataAsync(request);
            return Ok(response);
        }
        catch (ArgumentException ex)
        {
            _logger.LogWarning(ex, "Invalid request for saving template with AI metadata");
            return BadRequest(new { message = ex.Message });
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error saving template with AI metadata: {Error}", ex.Message);
            return StatusCode(500, "Internal server error");
        }
    }
}